libcalico-go/lib/epstatusfile/status_file_watcher.go

```diff
@@ -167,6 +164,9 @@ func (w *FileWatcher) runWatcher() {
 			log.WithError(err).Info("Error initializing fsnotify. Falling back to polling.")
 		}
 
+		// Get current state of the directory and emit initial events.
+		w.scanDirectory()
+		w.callbacks.OnInSync(true)
```

Comment:
I _think_ it'd be better to only call this once.  Downstream users getting the event might be confused by getting dupes, or attach meaning to it that we didn't intend.

Reply:
I believe I only left one call behind now

Reply:
I mean once total, not once per loop.

Reply:
Missed this one until now. I've wrapped the call in a Once